app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -127,9 +127,14 @@ class MainActivity : AppCompatActivity() {
 
     override fun onBackPressed() {
         val currentFragment = this.supportFragmentManager.findFragmentById(R.id.frameContainer)
+        val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
         if (currentFragment !is EventsFragment) {
-            loadFragment(EventsFragment())
-            navigation.selectedItemId = navigation_events
+            if (rootFragment is EventDetailsFragment) {
+                super.onBackPressed()
+            } else {
+                loadFragment(EventsFragment())
+                navigation.selectedItemId = navigation_events
+            }
```

Comment:
This is getting too complicated. Think of a different solution

Reply:
Updated.

Reply:
> This is getting too complicated. Think of a different solution

Updated. please review.